src/conduits/java/com/enderio/conduits/common/blocks/ConduitBlock.java

```diff
@@ -0,0 +1,391 @@
+                            }
+                        } else if (connectionState == StaticConnectionStates.DISCONNECTED) {
+                            conduit.tryConnectTo(direction, type, true, true);
+                        }
+                    }
+                }
+            }
+        }
+        super.neighborChanged(state, level, pos, block, fromPos, isMoving);
+    }
+
+    @Override
+    public PushReaction getPistonPushReaction(BlockState pState) {
+        return PushReaction.BLOCK;
+    }
+
+    @Override
+    public FluidState getFluidState(BlockState state) {
+        return state.getValue(WATERLOGGED) ? Fluids.WATER.getSource(false) : super.getFluidState(state);
+    }
+
+    @Nullable
+    @Override
+    public BlockEntity newBlockEntity(BlockPos pos, BlockState state) {
+        return ConduitBlockEntities.CONDUIT.create(pos, state);
+    }
+
+    @Override
+    public VoxelShape getShape(BlockState state, BlockGetter level, BlockPos pos, CollisionContext context) {
+        BlockEntity be = level.getBlockEntity(pos);
+        if (be instanceof ConduitBlockEntity conduit) {
+            return conduit.getShape().getTotalShape();
+        }
+        return Shapes.block();
+    }
+
+    @Override
+    public InteractionResult use(BlockState state, Level level, BlockPos pos, Player player, InteractionHand hand, BlockHitResult hit) {
+        BlockEntity be = level.getBlockEntity(pos);
+        if (be instanceof ConduitBlockEntity conduit) {
+            if (!player.getItemInHand(hand).isEmpty()) {
+                Optional<InteractionResult> interactionResult = addConduit(conduit, player, player.getItemInHand(hand), level.isClientSide());
+                if (interactionResult.isPresent())
+                    return interactionResult.get();
+                interactionResult = handleYeta(conduit, player, player.getItemInHand(hand), hit, level.isClientSide());
+                if (interactionResult.isPresent())
+                    return interactionResult.get();
+                interactionResult = handleFacade(conduit, player, player.getItemInHand(hand), hit, level.isClientSide());
+                if (interactionResult.isPresent())
+                    return interactionResult.get();
+            }
+            Optional<InteractionResult> interactionResult = handleScreen(conduit, player, hit, level.isClientSide());
+            if (interactionResult.isPresent())
+                return interactionResult.get();
+        }
+        return super.use(state, level, pos, player, hand, hit);
+    }
+
+    private Optional<InteractionResult> addConduit(ConduitBlockEntity conduit, Player player, ItemStack stack, boolean isClientSide) {
+        if (!(stack.getItem() instanceof ConduitBlockItem conduitBlockItem))
+            return Optional.empty();
+        EnderIO.LOGGER.info("rightclicked with conduititem: " + ConduitTypes.getRegistry().getKey(conduitBlockItem.getType()) + " @ " + conduit.getBlockPos().toShortString());
```

Comment:
Tbh we should do a PR reviewing all logger uses, because to me these should all be debug logs.